grafana/grafana
# Loki: Fix creating correct context query when preserved labels used

In https://github.com/grafana/grafana/pull/68700 I implemented logic to use `preserved labels` in `LokiContextUi`, but this should have been done in `LogContextProvider`, so both UI and created query use the same context labels. 

Because this was only in `LokiContextUi` the resulting query was incorrect because it didn't use preserved labels.

How to test:
1. Create loki query with parser
2. Open log context and remove some real labels and add some parsed labels (ideally `level`, so it is easy to observe correctness of the log lines)
3. Open new context for different row and observe if the log lines are correct, or you can even check network tab to make sure that resulting query is correct.


https://github.com/grafana/grafana/assets/30407135/c1e12509-a16c-4f73-a83e-4e8d7330a215

I am adding `no-changelog` label because this feature has not been released yet.
